### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by lesiadmytrus**
Added unit tests

### CODE REVIEW

File: src/app/components/student-form/student-form.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,192 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-
 import { StudentFormComponent } from './student-form.component';
+import { CommonModule } from '@angular/common';
+import { BrowserModule } from '@angular/platform-browser';
+import { ReactiveFormsModule, FormGroup, FormControl, Validators } from '@angular/forms';
+import { BsDatepickerModule } from 'ngx-bootstrap';
+import { HttpClientModule } from '@angular/common/http';
+import { RouterModule, Router} from '@angular/router';
```
Comment (dmutrokot) [inline src/app/components/student-form/student-form.component.spec.ts]: Seems that `Router` import is not neccessary

File: src/app/components/student-form/student-form.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,192 @@
...
+import { ToastrModule, ToastrService } from 'ngx-toastr';
```
Comment (dmutrokot) [inline src/app/components/student-form/student-form.component.spec.ts]: Seems that `ToastrService ` import is not neccessary

File: src/app/components/student-form/student-form.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,192 @@
...
+  firstName: new FormControl('ghgh', Validators.required),
+  lastName: new FormControl('ghmockStudenth', Validators.required),
+  email: new FormControl('mockStudenthgfh@ghf.gjff', Validators.required),
+  phoneNumber: new FormControl('+45445', Validators.required),
+  country: new FormControl('Italy', Validators.required),
+  dateOfBirth: new FormControl('2012-12-12', Validators.required),
+  gender: new FormControl()
+});
+
+const studentsMock: Student[] = [
```
Comment (dmutrokot) [inline src/app/components/student-form/student-form.component.spec.ts]: Do you use it?

File: src/app/components/student-form/student-form.component.spec.ts [hunk 4]
```diff
@@ -1,25 +1,192 @@
...
+      spyOn(messagesService, 'handlerSuccess');
+      component.update(mockStudent);
+      expect(messagesService.handlerSuccess).toHaveBeenCalledWith(mockSuccessMessage.message);
+    });
+  });
+
+  describe('GetStudent', () => {
+    it('#getStudent should get student by id', () => {
+      component.getStudent(mockStudentId);
+      expect(component.getStudent).toBeDefined();
```
Comment (dmutrokot) [inline src/app/components/student-form/student-form.component.spec.ts]: What this expect does do? :)

File: src/app/components/student/student.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    it('#getAll should set @isLoading "false"', () => {
+      const query = '';
+      component.getStudents(query);
+      expect(component.isLoading).toEqual(false);
+    });
+  });
+
+  describe('ConfirmDeletation', () => {
+    it('#delete should delete student', () => {
+      component.confirmDeletion();
```
Comment (dmutrokot) [inline src/app/components/student/student.component.spec.ts]: Maybe here should be some expect?

File: src/app/pipes/age.pipe.spec.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  describe('Calculations', () => {
+    it('should return false if invalid input data', () => {
+      const expectation = pipe.transform(invalidBirthdayInput);
+      expect(expectation).toBeFalsy();
+    });
+  
+    it('should transform input date to age', () => {
+      const expectation = pipe.transform(mockBirthday);
+      expect(expectation).toBe(8);
```
Comment (dmutrokot) [inline src/app/pipes/age.pipe.spec.ts]: Let's imagine that current date is 2019-12-12, and I suppose that, this test will be failed :)

File: src/app/components/student-form/student-form.component.spec.ts [hunk 7]
```diff
@@ -1,25 +1,192 @@
...
+    age: 120,
+    email: 'norman@g.com',
+    phoneNumber: 1234,
+    country: 'France',
+    gender: 'male',
+    dateOfBirth: '1899-10-29'
+  }
+];
+
+const emptyMockStudent = new FormGroup({});
```
Comment (vlodkopits) [inline src/app/components/student-form/student-form.component.spec.ts]: do you use it?

File: src/app/components/student/student.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+  {
+    field: 'firstName',
+    value: 'g'
+  }
+];
+
+describe('StudentComponent', () => {
+  let component: StudentComponent;
+  let fixture: ComponentFixture<StudentComponent>;
+  let studentService: MockStudentService;;
```
Comment (vlodkopits) [inline src/app/components/student/student.component.spec.ts]: extra `;`

File: src/app/components/student-form/student-form.component.spec.ts [hunk 9]
```diff
@@ -1,25 +1,192 @@
...
+  describe('AddStudent', () => {
+    it('#add should create new student', () => {
+      spyOn(messagesService, 'handlerSuccess');
+      component.add(mockStudent);
+      expect(messagesService.handlerSuccess).toHaveBeenCalledWith(mockSuccessMessage.message);
+    });
+
+    it('should check if "gender" is checked', () => {
+      component.add(mockStudentWithoutGender);
+      expect(component.add).toBeDefined();
```
Comment (vlodkopits) [inline src/app/components/student-form/student-form.component.spec.ts]: what do you check here?

File: src/app/components/spinner/spinner.component.ts [hunk 10]
```diff
@@ -1,13 +1,9 @@
...
   selector: 'app-spinner',
   templateUrl: './spinner.component.html',
   styleUrls: ['./spinner.component.scss']
 })
-export class SpinnerComponent implements OnInit {
-  constructor() { }
-
-  ngOnInit() {
-  }
+export class SpinnerComponent {
```
Comment (vlodkopits) [inline src/app/components/spinner/spinner.component.ts]: add an extra line to the end of file

File: src/app/components/student-form/student-form.component.spec.ts [hunk 11]
```diff
@@ -1,25 +1,193 @@
...
+          studentService = new MockStudentService();
+          messagesService = TestBed.get(MessagesService);
+      });
   }));
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(StudentFormComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+  xit('should be created', () => {
```
Comment (vlodkopits) [inline src/app/components/student-form/student-form.component.spec.ts]: ?

**Pull request merged**